src/components/hooks/useActivityScroll.js

```diff
@@ -0,0 +1,66 @@
+import {useContext, useEffect, useState} from 'react';
+import {empty, fromEvent} from 'rxjs';
+import {filter, flatMap} from 'rxjs/operators';
+
+import {AdapterContext} from '../../components/';
+
+/**
+ * Callback function to execute once data has been fetched.
+ *
+ * @callback previousActivitiesCallback
+ * @param {Array.<string|ActivityDate>} previousActivities Array of previous activities fetched.
+ * @returns undefined
+ */
+
+/**
+
+/**
+ * Custom hook returns a loading state to indicate when data is being fetched,
+ * once user has scrolled to the top of the component of the given DOM reference.
+ *
+ * @param {string} roomID  ID of the room from which to fetch data.
+ * @param {object} elementRef  reference to the element to attach scroll listener.
+ * @param {previousActivitiesCallback} callback  Callback to execute once data has been fetched.
+ * @returns {boolean}  Data loading state.
+ */
+export default function useActivityScroll(roomID, elementRef, callback) {
+  const {roomsAdapter} = useContext(AdapterContext);
+  const [showLoader, setShowLoader] = useState(false);
+
+  useEffect(() => {
+    let scrollListener = empty();
+
+    if (elementRef.current) {
+      // Listen to scroll event, and on reaching the top,
+      // And load previous activities if there are any
+      scrollListener = fromEvent(elementRef.current, 'scroll')
+        .pipe(
+          filter((event) => {
+            const isViewportTop = event.target.scrollTop === 0;
+            const loadMoreItems = isViewportTop && roomsAdapter.hasMoreActivities(roomID);
```

Comment:
When implementing the `hasMoreActivities` in the SDK adapter, we need to be sure to consider that there is no real way to know the room has or doesn't have more activities. The workaround would be that if the amount returned is less than the limit requested.

Reply:
really? From the SDK, there's no way to know that there is no more activities to load?

Reply:
Correct, take a look at the messages endpoint (https://developer.webex.com/docs/api/v1/messages/list-messages), all we get is an array and no meta-data about remaining messages 😢

Reply:
Oh that's fine, I think we should be able to know by not getting any more data (like receiving an empty array?). I guess we'll find out on the SDK adapter implementation.